managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImpl.java

```diff
@@ -398,10 +398,11 @@ private static ManagedLedgerInfo updateMLInfoTimestamp(ManagedLedgerInfo info) {
     }
 
     private static MetaStoreException getException(Throwable t) {
-        if (t.getCause() instanceof MetadataStoreException.BadVersionException) {
-            return new ManagedLedgerException.BadVersionException(t.getMessage());
+        Throwable actEx = FutureUtil.unwrapCompletionException(t);
+        if (actEx instanceof MetadataStoreException.BadVersionException) {
+            return new ManagedLedgerException.BadVersionException(actEx.getMessage());
```

Comment:
Why is a new exception created here? couldn't we just return the original instance `actEx`?

Reply:
One is `ManagedLedgerException. BadVersionException`, anther is `MetadataStoreException.BadVersionException`

Reply:
gotcha. How about wrapping the original exception instance with `return new ManagedLedgerException.BadVersionException(actEx)` instead of just wrapping the message? 
That would preserve stacktraces.

Reply:
Improved